packages/x-charts-pro/src/Heatmap/Heatmap.tsx

```diff
@@ -129,13 +144,26 @@ const Heatmap = React.forwardRef(function Heatmap(
   const clipPathId = `${id}-clip-path`;
 
   const defaultizedXAxis = React.useMemo(
-    () => xAxis.map((axis) => ({ scaleType: 'band' as const, categoryGapRatio: 0, ...axis })),
-    [xAxis],
+    () =>
+      (xAxis && xAxis.length > 0 ? xAxis : [{}]).map((axis) => ({
+        id: DEFAULT_X_AXIS_KEY,
+        scaleType: 'band' as const,
+        categoryGapRatio: 0,
+        data: axis.data ? undefined : getDefaultDataForXAxis(series),
+        ...axis,
```

Comment:
I was wondering why setting the data to `undefined` before realizing the real one comes from the `...axis`. Not sure it's clearer, but I let it as a proposal

```suggestion
        ...axis,
        data: axis.data ?? getDefaultDataForXAxis(series),
```